src/txn/txn_recover.c

```diff
@@ -1079,9 +1080,13 @@ __wt_txn_recover(WT_SESSION_IMPL *session, const char *cfg[])
     FLD_SET(conn->log_flags, WT_CONN_LOG_RECOVER_DONE);
 
     /* Time since the recovery has started. */
-    __wt_timer_evaluate(session, &timer, &time_diff);
+    __wt_timer_evaluate(session, &timer, &conn->recovery_timeline.recovery);
     __wt_verbose(session, WT_VERB_RECOVERY_PROGRESS,
-      "recovery has successfully finished and ran for %" PRIu64 " seconds", time_diff);
+      "recovery was completed successfully and took %" PRIu64 " ms, including %" PRIu64
+      " ms for the log replay, %" PRIu64 " ms the rollback to stable, and %" PRIu64
```

Comment:
The "for" has gone missing in a couple of spots here - e.g. "ms the rollback" -> "ms for the rollback". Plus the spaces thing from the other comment

Reply:
Fixed.